invenio_administration/menu/menu.py

```diff
@@ -83,10 +89,27 @@ def default_active_when(self):
 class MenuItem:
     """Class for menu item."""
 
-    def __init__(self, name="", endpoint="", category="", order=0, active_when=None):
+    def __init__(
+        self,
+        name="",
+        endpoint="",
+        category="",
+        order=0,
+        icon_key=None,
+        active_when=None
+    ):
         """Constructor."""
         self.name = name
         self.endpoint = endpoint
         self.category = category
         self.order = order
         self.active_when = active_when
+        self.icon_key = icon_key
+
+    @property
+    def icon(self):
+        """Return corresponding template path for icon."""
+        if not self.icon_key:
+            return None
+
+        return current_theme_icons[self.icon_key]
```

Comment:
What happens if `self.icon_key` is not a key in `current_theme_icons`?

Should this yield an error to warn the developer or should it render the menu entry without its icon?

Reply:
it will yield an error, since it is a dict

Reply:
`KeyError` with the default message is enough or shall we raise a `KeyError` with a custom message (and perhaps a hint)?

Reply:
I think it is enough for a developer